### ISSUE TITLE
Window going blank when content of realm server changes

### ISSUE THREAD

**Issue by programaths** [realm/realm-studio#1065]
**Describe the bug**

The windows first shows old content then goes blank

**To Reproduce**
1. Connect to a realm server using a JWT token
2. Open a realm
3. Close realm studio
4. Empty the realm
5. Open realm studio
6. Close realm studio
7. Fill the realm with some data
8. Open realm studio

Repeat steps 2 to 8 until bug

**Expected behavior**

**Versions:**
 - OS: Kubuntu (uname -a) Linux christian-Aspire-TC-605 4.15.0-43-generic #46-Ubuntu SMP Thu Dec 6 14:45:28 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
 - Studio Version : 3.2.0
 - Realm Object Server version: 3.16.5

**Additional context**

```
Failed when registering protocols
Studio could not register the x-realm-cloud:// protocol.
Failed when registering protocols
Studio could not register the x-realm-studio:// protocol.
Checking for update
Update for version 3.2.0 is not available (latest version: 3.2.0, downgrade is disallowed).
../../third_party/tcmalloc/chromium/src/tcmalloc.cc:289] Attempt to free invalid pointer 0x7f405bd763a0 
```

**Comment by programaths** [comment]
For others having the same problem, you can delete the content of:

`/$HOME/.config/Realm Studio/renderer-server-administration/realm-object-server`

Then reconnect and enjoy!

**Comment by programaths** [comment]
I suspect that "clear caches" should clear those folders, seems it doesn't.

Had issues on a realm itself and deleted `renderer-*` to get it working again.

**Comment by kraenhansen** [comment]
Thanks for reporting this!
To reproduce it, whey you write "Empty the realm" what do you do exactly? Do you do this from an app or script external to Studio?

**Comment by programaths** [comment]
It's even simpler:
I recreate the server!
For step 3, closing realm studio completely or keeping the main window only does the same thing.

So, here are simpler steps:
1. Initialize a ROS server in folder A
2. Feed data into it through an external client (NodeJS Script)
3. Connect with Realm-Studio
4. Disconnect
5. Initialize a ROS server in folder B
6. Feed data into it through an external client (NodeJS Script)
7. Connect with Realm-Studio

The window will open, show content then become blank.

**Comment by kraenhansen** [comment]
I suspect this happens because Realm Studio is acting as a client and [keeps a local copy](https://github.com/realm/realm-studio/pull/970) of the /__admin Realm. This will result in an unhandled client reset error when opening a server with the same hostname but different data directory. Studio should handle this reset gracefully by initiating a client reset when the error is thrown.

**Comment by nirinchev** [comment]
It [does handle the client reset error](https://github.com/realm/realm-studio/blob/master/src/services/ros/realms.ts#L76-L80) when opening the Realm, so it's not clear to me why this would happen assuming the relevant windows are closed between changes.

**Comment by kraenhansen** [comment]
@nirinchev the server administration window doesn't call that function to open Realms anymore, it was refactored to use [react-realm-context](https://www.npmjs.com/package/react-realm-context) instead.

**Comment by cyphire** [comment]
> For others having the same problem, you can delete the content of:
> 
> `/$HOME/.config/Realm Studio/renderer-server-administration/realm-object-server`
> 
> Then reconnect and enjoy!

SAVED MY LIFE!!! Amazing how realm.io has no real documentation nor any feedback for this type of thing.  Could not get my always working Realm Studio working until I deleted my:

For others having the same problem, you can delete the content of:

/$HOME/.config/Realm Studio/renderer-server-administration/realm-object-server

or in my case (on WIndows 10): 

C:\Users\<myaccount>\AppData\Roaming\Realm Studio\renderer-server-administration\realm-object-server

Thank you!  AND SOMEONE TELL REALM!!!

**Comment by programaths** [comment]
@cyphire They are well aware and when I worked with their technology, I did even testing for them...for free!

Then they have been baught, so I expected it to go better.

Good luck with it.

**Comment by cyphire** [comment]
Thanks so much.  I just mean that things like this shouldn't take 2 days of github searches.  They need to have real troubleshooting guides and the site should have given this answer.  Glad that you feel the product will get better with their purchase by Mongo.  Also just to mention, I have been unable to do real debugging as their product is embedded in our app.  I use Reactotron to give me state and logging, but because of the 20-40ms bridge between native and js, debugging is about impossible.  I hope that they fix this issue, or give a way of doing debugging by maybe not polling so often, it would make future use of the product important as I am so close to not using realm any more and in search of a replacement.

On the positive side since I haven't had debugging in 2 years, it made me a much better programmer!

**Comment by programaths** [comment]
@cyphire Well, I see that in reality it didn't evolve much AND they cleaned most of the issues I had in the paste.

If you want to have a laugh, have a look at that: https://github.com/realm/realm-js/issues/2181#issuecomment-521681427

Or maybe you won't.

Good luck!

**Comment by kraenhansen** [comment]
@cyphire

> SAVED MY LIFE!!! Amazing how realm.io has no real documentation nor any feedback for this type of thing.

I believe the primary reason this is not documented, is because it's an error happening at very rare occasions (required deleted Realms or resetting the server). I think we should spend time fixing issues instead of bloating the documentation with guides involving weird work-arounds for rare issues. But thats just my opinion.

> Thank you! AND SOMEONE TELL REALM!!!

We're right here 🙂

> I just mean that things like this shouldn't take 2 days of github searches.

Next time you experience something similar, feel free to create an issue. We'll do our very best at pointing you to know work-arounds, existing known issues, etc. We tend to respond within 2 days - but as I can see on @programaths's, some issues fall through the cracks. If that happens - feel free to ping us 👍

---

We recently merged a fix for client resets that might have fixed the original issue (https://github.com/realm/realm-studio/pull/1238), it got released with Realm Studio 3.9.0. I suppose @programaths doesn't have an incentive to test if this fixed his original issue, but perhaps I could get you @cyphire to check if the latest Realm Studio has this issue and if it does, create a new issue with a step-by-step list instructions we can follow to reproduce it.

We really want to get to the bottom of issues like this!
